### ISSUE TITLE
Text line inputs crash when encountering non-ASCII text.

### ISSUE THREAD

**Issue by Evrey** [lapce/lapce#84]
## How to replicate:

Type `»«` into the command palette. Attempt to delete the `«` by hitting backspace.

This likely affects every single line input.
```
thread 'main' panicked at 'byte index 3 is not a char boundary; it is inside '«' (bytes 2..4) of `»«`', /rustc/e7aca895980f25f6d2d3c48e10fd04656764d1e4/library/alloc/src/string.rs:1259:24
stack backtrace:
   0: rust_begin_unwind
             at /rustc/e7aca895980f25f6d2d3c48e10fd04656764d1e4/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/e7aca895980f25f6d2d3c48e10fd04656764d1e4/library/core/src/panicking.rs:143:14
   2: core::str::slice_error_fail
   3: <lapce_core::palette::PaletteViewData as lapce_core::keypress::KeyPressFocus>::run_command
   4: lapce_core::keypress::KeyPressData::run_command
   5: lapce_core::keypress::KeyPressData::key_down
   6: <lapce_core::palette::NewPalette as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
   7: druid::core::WidgetPod<T,W>::event
   8: <lapce_core::tab::LapceTabNew as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
   9: <lapce_core::data::LapceTabLens as druid::lens::lens::Lens<lapce_core::data::LapceWindowData,lapce_core::data::LapceTabData>>::with_mut
  10: <druid::widget::lens_wrap::LensWrap<T,U,L,W> as druid::widget::widget::Widget<T>>::event
  11: druid::core::WidgetPod<T,W>::event
  12: <lapce_core::window::LapceWindowNew as druid::widget::widget::Widget<lapce_core::data::LapceWindowData>>::event
  13: <lapce_core::data::LapceWindowLens as druid::lens::lens::Lens<lapce_core::data::LapceData,lapce_core::data::LapceWindowData>>::with_mut
  14: <druid::widget::lens_wrap::LensWrap<T,U,L,W> as druid::widget::widget::Widget<T>>::event
  15: druid::core::WidgetPod<T,W>::event
  16: druid::window::Window<T>::event
  17: druid::win_handler::InnerAppState<T>::do_window_event
  18: druid::win_handler::AppState<T>::do_window_event
  19: druid::win_handler::AppState<T>::do_winit_window_event
  20: druid::app::AppLauncher<T>::launch::{{closure}}
  21: winit::platform_impl::platform::wayland::event_loop::EventLoop<T>::run
  22: winit::platform_impl::platform::EventLoop<T>::run
  23: winit::event_loop::EventLoop<T>::run
  24: druid::app::AppLauncher<T>::launch
  25: lapce_core::app::lanuch
```
Also breaks e.g. the theme setting:
```
thread 'main' panicked at 'byte index 14 is not a char boundary; it is inside '«' (bytes 13..15) of `Lapce Light»«`', /rustc/e7aca895980f25f6d2d3c48e10fd04656764d1e4/library/alloc/src/string.rs:1259:24
stack backtrace:
   0: rust_begin_unwind
             at /rustc/e7aca895980f25f6d2d3c48e10fd04656764d1e4/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/e7aca895980f25f6d2d3c48e10fd04656764d1e4/library/core/src/panicking.rs:143:14
   2: core::str::slice_error_fail
   3: <lapce_core::settings::LapceSettingsItemKeypress as lapce_core::keypress::KeyPressFocus>::run_command
   4: lapce_core::keypress::KeyPressData::run_command
   5: lapce_core::keypress::KeyPressData::key_down
   6: <lapce_core::settings::LapceSettingsItem as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
   7: druid::core::WidgetPod<T,W>::event
   8: druid::core::WidgetPod<T,W>::event
   9: <lapce_core::settings::LapceSettings as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
  10: druid::core::WidgetPod<T,W>::event
  11: <lapce_core::scroll::LapceScrollNew<T,W> as druid::widget::widget::Widget<T>>::event
  12: druid::core::WidgetPod<T,W>::event
  13: <lapce_core::split::LapceSplitNew as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
  14: druid::core::WidgetPod<T,W>::event
  15: <lapce_core::settings::LapceSettingsPanel as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
  16: druid::core::WidgetPod<T,W>::event
  17: <lapce_core::tab::LapceTabNew as druid::widget::widget::Widget<lapce_core::data::LapceTabData>>::event
  18: <lapce_core::data::LapceTabLens as druid::lens::lens::Lens<lapce_core::data::LapceWindowData,lapce_core::data::LapceTabData>>::with_mut
  19: <druid::widget::lens_wrap::LensWrap<T,U,L,W> as druid::widget::widget::Widget<T>>::event
  20: druid::core::WidgetPod<T,W>::event
  21: <lapce_core::window::LapceWindowNew as druid::widget::widget::Widget<lapce_core::data::LapceWindowData>>::event
  22: <lapce_core::data::LapceWindowLens as druid::lens::lens::Lens<lapce_core::data::LapceData,lapce_core::data::LapceWindowData>>::with_mut
  23: <druid::widget::lens_wrap::LensWrap<T,U,L,W> as druid::widget::widget::Widget<T>>::event
  24: druid::core::WidgetPod<T,W>::event
  25: druid::window::Window<T>::event
  26: druid::win_handler::InnerAppState<T>::do_window_event
  27: druid::win_handler::AppState<T>::do_window_event
  28: druid::win_handler::AppState<T>::do_winit_window_event
  29: druid::app::AppLauncher<T>::launch::{{closure}}
  30: winit::platform_impl::platform::wayland::event_loop::EventLoop<T>::run
  31: winit::platform_impl::platform::EventLoop<T>::run
  32: winit::event_loop::EventLoop<T>::run
  33: druid::app::AppLauncher<T>::launch
  34: lapce_core::app::lanuch
```
After program restart:
![image](https://user-images.githubusercontent.com/6483505/156568959-779fbe7d-1f17-4335-817e-6447b3613abe.png)
The theme setting is now broken and irreparable unless one was to touch the config files directly.

**Comment by bugadani** [comment]
Thanks for reporting this issue! I can confirm it is still present on the `master` branch. I believe this shouldn't be hard to fix, so I'm adding the `good first issue` label in case someone wants to pick it up.

**Comment by dzhou121** [comment]
This should be fixed now by the latest v0.1.0 release.
